book/website/reproducible-research/case-studies/statistical-methods-manuscript.md

```diff
@@ -0,0 +1,146 @@
+To execute this workflow,
+simply open a shell in the online version of Jupyterhub (via the binderlink)
+and type
+```
+snakemake -F --cores 1  manuscript
+```
+This will execute all required steps in turn, i.e., first create all plots 
+and then compiling the actual `latex/main.pdf` file from the LaTeX sources.
+All relevant configurations for binder ate located in `.binder`.
+
+
+
+### Support for local instantiation of the software environment
+
+The python package repo2docker can also be used locally to reproduce the exact 
+same computing environment.
+To this end you will need to have python and docker installed. 
+For details on docker and container technologies in general, see {ref}`rr:renv:containers`.
+Then simply clone the repository on your local machine via
+```
+git clone git@github.com:kkmann/sample-size-calculation-under-uncertainty.git
+cd sample-size-calculation-under-uncertainty
+```
+and  build a docker container from the configuration files in `.binder/` via
+```
+jupyter-repo2docker -E .
+```
+after cloning the repository.
+The container is started automatically after the build completes and you can
+use the usual Jupyter interface in you browser (via the link printed by repo2docker)
+to explore the repository locally.
+
+
+### Use of cntinuous integration
+
+Although not absolutely necessary from a reproducibility perspective,
+the repository also makes use of continuous integration (CE, see {ref}`rr:ci`)
+via [GitHub actions](https://github.com/features/actions).
+GitHub actions are similar in sprit to {ref}`rr:ci:travis` but the runners are
+provided directly via GitHub.
+The repository defines two workflows in `.github/workflows`.
+The first one, [`.github/workflows/build_and_run.yml`](https://github.com/kkmann/sample-size-calculation-under-uncertainty/blob/master/.github/workflows/build_and_run.yml),
+is activated whenever the master branch of the repository is updated and the specifications ins `.binder` are changed,
+builds the container, pushes it to a public container repository [docker hub](https://hub.docker.com/repository/docker/kkmann/sample-size-calculation-under-uncertainty) and then checks that the snakemake workflow runs through without problems.
+The second one, [`.github/workflows/run.yml`](https://github.com/kkmann/sample-size-calculation-under-uncertainty/blob/master/.github/workflows/run.yml),
+runs when `.binder` was not changed and uses the prebuild docker container
+to run the snakemake workflow.
+The latter saves a lot of computing time (and this trees) since the comoutational
+environment will change much less often than the contents of the repository. 
+The use of CI thus facilitates checking contributions via pull requests for
+technical integrity and makes the respective latest version of the required container 
+available for direct download. 
+I.e., instead of building the container locally via repo2docker you could just 
+download it directly and execute the workflow via 
```

Comment:
```suggestion
The second one, [`.github/workflows/run.yml`](https://github.com/kkmann/sample-size-calculation-under-uncertainty/blob/master/.github/workflows/run.yml),
runs with an unchanged `.binder` using the pre-built Docker container
to run the snakemake workflow.
The latter saves a lot of computing time (and this trees) since the computational 
environment will change much less often than the contents of the repository. 
The use of CI thus facilitates checking contributions via pull requests for
technical integrity and makes the respective latest version of the required container 
available for direct download. 
This means that instead of building the container locally via repo2docker you could 
download it directly and execute the workflow using the following commands:
```

Reply:
what does "(and this trees)" mean?

Reply:
'thus', a failed attempt at humor x)